js/main.js

```diff
@@ -6,72 +6,263 @@ var proName;
+var sortArr = [];
+//defult data show
+getStoreData(0)
+//form validation
+function validateform(id) {
+    getFormValue();
+    var allowedExtensions = /(\.jpg|\.jpeg|\.png|\.jfif|\.gif)$/i;
+    if (isNaN(proId) || proId < 1) {
+        document.getElementById('errorMessageId').innerText = "Please input a number only!!";
+    }
+    else if (proName == null || proName == "") {
+        document.getElementById('errorMessageName').innerText = "Please input a product name!!";
+    }
+    else if (!allowedExtensions.exec(proImage)) {
+        document.getElementById('errorMessageImage').innerText = "File type is not allowed!!";
+        proImage.value = '';
+        return false;
+    }
+    else if (isNaN(proPrice) || proPrice < 1) {
+        document.getElementById('errorMessagePrice').innerText = "Please input a number only!!";
+    }
+    else {
+        if (id == 'add') {
+            formData();
+            alert('Product added successfull!!');
+        }
+        else {
+            setEditdata();
+        }
+    }
+}
+//store data function
 function formData() {
     getFormValue();
     createArray();
     storeData();
 }
-
+//get form value function
 function getFormValue() {
     proId = document.getElementById('productId').value;
     proName = document.getElementById('ProductName').value;
-    proImage = document.getElementById('image').value;
+    proImage = document.getElementById('image').files[0].name;
     proPrice = document.getElementById('price').value;
     proDecpt = document.getElementById('description').value;
 }
-
+//create array function
 function createArray() {
     proArray.push(proId, proName, proImage, proPrice, proDecpt);
 }
-
+//store data function
 function storeData() {
+    let arrayJSON = JSON.stringify(proArray);
     getVal = btnValueGet();
-    //console.log(getVal);
     if (getVal == null) {
-        localStorage.setItem('productValue0', JSON.stringify(proArray));
+        localStorage.setItem('productValue0', arrayJSON);
         getVal += 1;
         btnValueSet(getVal);
     }
     else {
         let pro = "productValue" + getVal;
-        localStorage.setItem(pro, JSON.stringify(proArray));
+        localStorage.setItem(pro, arrayJSON);
         getVal += 1;
         btnValueSet(getVal);
     }
-
 }
-
-function getStoreData(){
+function validate() { 
     getVal = btnValueGet();
-    var table = document.getElementById("proDetails");
-    for(let i=0;i<getVal;i++){
-        let pro = "productValue" + i;
-        var row = table.insertRow(i);
+    proId = document.getElementById('productId').value;
+    proName = document.getElementById('ProductName').value;
+    for (let i = 0; i < getVal; i++) {
+        pro = "productValue" + i;
         var data = JSON.parse(localStorage.getItem(pro));
-        for (let j = 0 ;j < data.length+1 ;j++ )
+        console.log(data[0]);
+        if(proId == data[0] || proName == data[1])
         {
+            let msg = "Product ID and Name should be unique"; 
```

Comment:
This message and check meaning less. Suppose if i will add unique Id but same name then again i can see same error message. Is that valid?